File: backend/src/main/java/com/team05/petmeeting/domain/comment/dto/AnimalCommentRes.java
package com.team05.petmeeting.domain.comment.dto;

import com.team05.petmeeting.domain.comment.entity.AnimalComment;

import java.time.LocalDateTime;

public record AnimalCommentRes(
        Long userId,
        String nickname,
        String profileImageUrl,
        Long commentId,
        String content,
        Long feedId,
        LocalDateTime createdAt
) {
    public static AnimalCommentRes from(AnimalComment comment) {
        return new AnimalCommentRes(
                comment.getUser().getId(),
                comment.getUser().getNickname(),
                comment.getUser().getProfileImageUrl(),
                comment.getId(),
                comment.getContent(),
                comment.getAnimal().getId(),
                comment.getCreatedAt()
        );
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/comment/dto/AnimalCommentRes.kt
package com.team05.petmeeting.domain.comment.dto

import com.team05.petmeeting.domain.comment.entity.AnimalComment
import java.time.LocalDateTime

data class AnimalCommentRes(
    val userId: Long,
    val nickname: String,
    val profileImageUrl: String?,
    val commentId: Long,
    val content: String,
    val feedId: Long,
    val createdAt: LocalDateTime
) {
    companion object {
        @JvmStatic
        fun from(comment: AnimalComment): AnimalCommentRes {
            return AnimalCommentRes(
                comment.user.id,
                comment.user.nickname,
                comment.user.profileImageUrl,
                comment.id,
                comment.content,
                comment.animal.id,
                comment.createdAt
            )
        }
    }
}
